### ISSUE TITLE
Update appstore to list Nextcloud 20 compatibility

### ISSUE THREAD

**Issue by jkahrs** [nextcloud/cookbook#419]
**Is your feature request related to a problem? Please describe.**
Using the Nextcloud store Website and API won't list a compatible release for nextcloud 20.

**Describe the solution you'd like**
The currently released version seems to be running on nextlcloud 20 and should be published to support the new version

**Comment by christianlupus** [comment]
Duplicate of #331
Slightly related to #233

**Comment by Zocker1999NET** [comment]
@christianlupus This is still an issue as for most users the app is not listed as compatible to version 20. If this issue is a duplicate to #331 in your opinion, then please reopen #331 as this issue is not resolved for the common user base.

Also, it might be related to #223, but reading through the comments does not give me the expectation that the automatic deployment will be available soon (speaking of 2 or 3 days), which is okay, but, in my opinion, then an official release should be uploaded soon instead of waiting for the automatic deployment, resulting in that either this issue or #331 should still be open until a NC 20 compatible release hits the store.

**Comment by christianlupus** [comment]
I"ll reopen here, until the CI is running.

**Comment by christianlupus** [comment]
@mrzapp If you are able to create a release manually before the weekend, this would relax the situation a bit. Apart from that, I plan to implement something regarding the CI on the weekend.

**Comment by mrzapp** [comment]
@christianlupus sure thing, but the `stable` branch is still `v0.7.6`, which is in the app store. Is it safe to merge `master` into `stable` at this point?

**Comment by mrzapp** [comment]
Done :)

**Comment by christianlupus** [comment]
@mrzapp What makes you consider a merge unsafe?

**Comment by mrzapp** [comment]
@christianlupus I got a couple errors:

```
[index] Error: Doctrine\DBAL\Exception\NotNullConstraintViolationException: An exception occurred while executing 'INSERT INTO "oc_preferences" ("userid", "appid", "configkey", "configvalue") VALUES(?, ?, ?, ?)' with params [null, "cookbook", "last_index_update", 1607713219]:

SQLSTATE[23502]: Not null violation: 7 ERROR:  null value in column "userid" violates not-null constraint
DETAIL:  Failing row contains (null, cookbook, last_index_update, 1607713219). at <<closure>>

 0. /var/www/html/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php line 169
    Doctrine\DBAL\Driver\AbstractPostgreSQLDriver->convertException("An exception oc ... .", Doctrine\DBAL\Dr ... ]})
 1. /var/www/html/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php line 149
    Doctrine\DBAL\DBALException::wrapException(Doctrine\DBAL\Driver\PDOPgSql\Driver {}, Doctrine\DBAL\Dr ... ]}, "An exception oc ... .")
 2. /var/www/html/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Connection.php line 1071
    Doctrine\DBAL\DBALException::driverExceptionDuringQuery(Doctrine\DBAL\Driver\PDOPgSql\Driver {}, Doctrine\DBAL\Dr ... ]}, "INSERT INTO \"o ... )", {1: null,2: "coo ... 9})
 3. /var/www/html/lib/private/DB/Connection.php line 222
    Doctrine\DBAL\Connection->executeUpdate("INSERT INTO \"o ... )", [null,"cookbook" ... 9], [2,2,2,1])
 4. /var/www/html/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Query/QueryBuilder.php line 209
    OC\DB\Connection->executeUpdate("INSERT INTO \"o ... )", {dcValue1: null, ... 9}, {dcValue1: 2,dcV ... 1})
 5. /var/www/html/lib/private/DB/QueryBuilder/QueryBuilder.php line 217
    Doctrine\DBAL\Query\QueryBuilder->execute()
 6. /var/www/html/lib/private/DB/Connection.php line 300
    OC\DB\QueryBuilder\QueryBuilder->execute()
 7. /var/www/html/lib/private/AllConfig.php line 292
    OC\DB\Connection->setValues("preferences", {userid: null,ap ... "}, {configvalue: 1607713219}, [])
 8. /var/www/html/custom_apps/cookbook/lib/Service/DbCacheService.php line 341
    OC\AllConfig->setUserValue(null, "cookbook", "last_index_update", 1607713219)
 9. /var/www/html/custom_apps/cookbook/lib/Service/DbCacheService.php line 327
    OCA\Cookbook\Service\DbCacheService->checkSearchIndexUpdate()
10. /var/www/html/custom_apps/cookbook/lib/Controller/MainController.php line 46
    OCA\Cookbook\Service\DbCacheService->triggerCheck()
11. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 169
    OCA\Cookbook\Controller\MainController->index()
12. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 100
    OC\AppFramework\Http\Dispatcher->executeController(OCA\Cookbook\Controller\MainController {}, "index")
13. /var/www/html/lib/private/AppFramework/App.php line 152
    OC\AppFramework\Http\Dispatcher->dispatch(OCA\Cookbook\Controller\MainController {}, "index")
14. /var/www/html/lib/private/Route/Router.php line 308
    OC\AppFramework\App::main("OCA\\Cookbook\\ ... r", "index", OC\AppFramework\ ... {}, {action: null,_r ... "})
15. /var/www/html/lib/base.php line 1008
    OC\Route\Router->match("/apps/cookbook/")
16. /var/www/html/index.php line 37
    OC::handleRequest()

GET /apps/cookbook/
```

```
[PHP] Error: Error: Undefined property: OCA\Cookbook\Service\DbCacheService::$user_id at /var/www/html/custom_apps/cookbook/lib/Service/DbCacheService.php#341 at <<closure>>

0. /var/www/html/custom_apps/cookbook/lib/Service/DbCacheService.php line 341
   OC\Log\ErrorHandler::onError(8, "Undefined prope ... d", "/var/www/html/c ... p", 341, {last_index_update: 0,interval: 5})
1. /var/www/html/custom_apps/cookbook/lib/Service/DbCacheService.php line 327
   OCA\Cookbook\Service\DbCacheService->checkSearchIndexUpdate()
2. /var/www/html/custom_apps/cookbook/lib/Controller/MainController.php line 46
   OCA\Cookbook\Service\DbCacheService->triggerCheck()
3. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 169
   OCA\Cookbook\Controller\MainController->index()
4. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 100
   OC\AppFramework\Http\Dispatcher->executeController(OCA\Cookbook\Controller\MainController {}, "index")
5. /var/www/html/lib/private/AppFramework/App.php line 152
   OC\AppFramework\Http\Dispatcher->dispatch(OCA\Cookbook\Controller\MainController {}, "index")
6. /var/www/html/lib/private/Route/Router.php line 308
   OC\AppFramework\App::main("OCA\\Cookbook\\ ... r", "index", OC\AppFramework\ ... {}, {action: null,_r ... "})
7. /var/www/html/lib/base.php line 1008
   OC\Route\Router->match("/apps/cookbook/")
8. /var/www/html/index.php line 37
   OC::handleRequest()
```

```
[PHP] Error: Error: trim() expects parameter 1 to be string, array given at /var/www/html/custom_apps/cookbook/lib/Service/DbCacheService.php#261 at <<closure>>

 0. <<closure>>
    OC\Log\ErrorHandler::onError(2, "trim() expects  ... n", "/var/www/html/c ... p", 261, {json: {@context ... }})
 1. /var/www/html/custom_apps/cookbook/lib/Service/DbCacheService.php line 261
    trim(["Main Course"])
 2. /var/www/html/custom_apps/cookbook/lib/Service/DbCacheService.php line 234
    OCA\Cookbook\Service\DbCacheService->hasJSONCategory({@context: "http ... 9})
 3. /var/www/html/custom_apps/cookbook/lib/Service/DbCacheService.php line 86
    OCA\Cookbook\Service\DbCacheService->updateCategories("*** sensitive parameters replaced ***")
 4. /var/www/html/custom_apps/cookbook/lib/Service/DbCacheService.php line 50
    OCA\Cookbook\Service\DbCacheService->carryOutUpdate()
 5. /var/www/html/custom_apps/cookbook/lib/Service/DbCacheService.php line 338
    OCA\Cookbook\Service\DbCacheService->updateCache("*** sensitive parameters replaced ***")
 6. /var/www/html/custom_apps/cookbook/lib/Service/DbCacheService.php line 327
    OCA\Cookbook\Service\DbCacheService->checkSearchIndexUpdate()
 7. /var/www/html/custom_apps/cookbook/lib/Controller/MainController.php line 46
    OCA\Cookbook\Service\DbCacheService->triggerCheck()
 8. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 169
    OCA\Cookbook\Controller\MainController->index()
 9. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 100
    OC\AppFramework\Http\Dispatcher->executeController(OCA\Cookbook\Controller\MainController {}, "index")
10. /var/www/html/lib/private/AppFramework/App.php line 152
    OC\AppFramework\Http\Dispatcher->dispatch(OCA\Cookbook\Controller\MainController {}, "index")
11. /var/www/html/lib/private/Route/Router.php line 308
    OC\AppFramework\App::main("OCA\\Cookbook\\ ... r", "index", OC\AppFramework\ ... {}, {action: null,_r ... "})
12. /var/www/html/lib/base.php line 1008
    OC\Route\Router->match("/apps/cookbook/")
13. /var/www/html/index.php line 37
    OC::handleRequest()

GET /apps/cookbook/
```

And the "internal server error" on the app page. Not sure if this is related to NC20

**Comment by christianlupus** [comment]
@mrzapp I merged fix #434 that should handle these errors. Could you please retest?

Apart from that, I am a bit surprised that the error did not happen on my dev installation. I found that I needed to increase verbosity a bit, eventually.

**Comment by mrzapp** [comment]
@christianlupus that does indeed seem to fix it, nice :+1: 

Closing this BTW, since the app is now listed as NC20 compatible
